### PR TITLE
Fix issue with getting ownership on Windows

### LINKED ISSUE #5019: `panic: The system cannot find the file specified` when installing Elastic Agent on AWS EC2 Windows Server
Please post all questions and issues concerning the Elastic Agent on https://discuss.elastic.co/c/beats
before opening a Github Issue. Your questions will reach a wider audience there,
and if we confirm that there is a bug, then you can open a new issue.

For security vulnerabilities please only send reports to security@elastic.co.
See https://www.elastic.co/community/security for more information.

Please include configurations and logs if available.

For confirmed bugs, please report:
- Version: 8.14.x
- Operating System: Windows Server 2022
- Discuss Forum URL: N/A (Internal Slack communication, asked to make issue by @cmacknz )
- Steps to Reproduce: 

We are currently setting up a Windows EC2 instance in our AWS environment to use Elastic Agent to report back on our new ECS cluster. We were having issues setting up the Elastic Agent on the node, and @anuragsoni tried to manually bring up a Windows Server 2022 EC2 instance (AMI: `Windows_Server-2022-English-Full-ECS_Optimized`) for debugging. He logged in using the session manager via AWS, & followed the install steps given by Fleet (relevant fields redacted):

```
$ProgressPreference = 'SilentlyContinue'
Invoke-WebRequest -Uri https://artifacts.elastic.co/downloads/beats/elastic-agent/elastic-agent-8.14.1-windows-x86_64.zip -OutFile elastic-agent-8.14.1-windows-x86_64.zip
Expand-Archive .\elastic-agent-8.14.1-windows-x86_64.zip -DestinationPath .
cd elastic-agent-8.14.1-windows-x86_64
.\elastic-agent.exe install --url=REDACT --enrollment-token=REDACT
```

Upon running the install, the following log is generated:

```
Elastic Agent will be installed at C:\Program Files\Elastic\Agent and will run as a service. Do you want to continue? [Y/n]:y
[    ] Installing Elastic Agent...  [0s] panic: The system cannot find the file specified.

goroutine 1 [running]:
github.com/elastic/elastic-agent/pkg/utils.CurrentFileOwner()
        github.com/elastic/elastic-agent/pkg/utils/perm_windows.go:37 +0x45
github.com/elastic/elastic-agent/internal/pkg/agent/perms.newOpts({0xc0007bf030, 0x1, 0xc0007bede8?})
        github.com/elastic/elastic-agent/internal/pkg/agent/perms/opts.go:41 +0x25
github.com/elastic/elastic-agent/internal/pkg/agent/perms.FixPermissions({0xc00044db40, 0x29}, {0xc0007bf030?, 0x31556a0?, 0xc000830780?})
        github.com/elastic/elastic-agent/internal/pkg/agent/perms/windows.go:25 +0x45
github.com/elastic/elastic-agent/internal/pkg/agent/install.fixInstallMarkerPermissions({0xc00044db40?, 0x29?}, {{0x0?, 0xc00004f4e0?}, {0x0?, 0xc0000001e8?}})
        github.com/elastic/elastic-agent/internal/pkg/agent/install/install_windows.go:52 +0x77
github.com/elastic/elastic-agent/internal/pkg/agent/install.CreateInstallMarker({0xc00004f4e0?, 0x1e?}, {{0x0?, 0xc0007bf130?}, {0x0?, 0x19effdd64f0?}})
        github.com/elastic/elastic-agent/internal/pkg/agent/install/install.go:519 +0xd6
github.com/elastic/elastic-agent/internal/pkg/agent/install.setupInstallPath({0xc00004f4e0, 0x1e}, {{0x0?, 0xa?}, {0x0?, 0x0?}})
        github.com/elastic/elastic-agent/internal/pkg/agent/install/install.go:243 +0x405
github.com/elastic/elastic-agent/internal/pkg/agent/install.Install({0xc000103860, 0x38}, {0xc00004f4e0, 0x1e}, 0x0, 0x9?, 0x374e5ee?, 0x6?)
        github.com/elastic/elastic-agent/internal/pkg/agent/install/install.go:106 +0xcdc
github.com/elastic/elastic-agent/internal/pkg/agent/cmd.installCmd(0xc000718c90, 0x0?)
        github.com/elastic/elastic-agent/internal/pkg/agent/cmd/install.go:203 +0x121f
github.com/elastic/elastic-agent/internal/pkg/agent/cmd.newInstallCommandWithArgs.func1(0xc0004f9100?, {0x374c174?, 0x4?, 0x374c078?})
        github.com/elastic/elastic-agent/internal/pkg/agent/cmd/install.go:40 +0x26
github.com/spf13/cobra.(*Command).execute(0xc000005200, {0xc000716f20, 0x2, 0x2})
        github.com/spf13/cobra@v1.8.0/command.go:987 +0xaa3
github.com/spf13/cobra.(*Command).ExecuteC(0xc000004300)
        github.com/spf13/cobra@v1.8.0/command.go:1115 +0x3ff
github.com/spf13/cobra.(*Command).Execute(0xc000140040?)
        github.com/spf13/cobra@v1.8.0/command.go:1039 +0x13
main.main()
        github.com/elastic/elastic-agent/main.go:41 +0x233
PS C:\elastic-agent-8.14.1-windows-x86_64>
```

We believe the following code is to blame as the 8.13.4 install works just fine:

https://github.com/elastic/elastic-agent/blob/1348b9ca7bff4d9bb7b0507e2524fbdaa8a85358/pkg/utils/perm_windows.go#L28-L43

(Possible related issue: https://github.com/golang/go/issues/37348)

At this time in the install script, we believe that there is no user setup with a home directory. This may also be the case when we're running the install steps, as these are included in the launch template for the machine. While we may need to do some user setup, this panic behaviour is not ideal!

### PR DESCRIPTION
**Pull request by blakerouse**
## What does this PR do?

Fixes issue with `utils.CurrentFileOwner()` where it would panic. Now it returns an error and no longer needs to get the current home directory of the user.

## Why is it important?

Fixes an issue where installation can fail in the case that the executing user doesn't have a home directory.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works~~
- [x] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)
- ~~[ ] I have added an integration test or an E2E test~~

## Related issues

- Closes #5019

### CODE REVIEW

**Review by leehinman — APPROVED** [review]
LGTM.  Once suggestion on a breadcrumb for our future selves.

File: pkg/utils/perm_windows.go [hunk 1]
```diff
@@ -30,16 +31,34 @@ type FileOwner struct {
 // Note: Very unlikely for this to panic if this function is unable to get the current
 // user. Not being able to get the current user, is a critical problem and nothing
 // can continue so a panic is appropriate.
-func CurrentFileOwner() FileOwner {
-	u, err := user.Current()
+func CurrentFileOwner() (FileOwner, error) {
+	t, err := syscall.OpenCurrentProcessToken()
```
[the remark this answers is not in the document]
Comment (blakerouse) [reply pkg/utils/perm_windows.go:39]: Added!

**Review by michalpristas — APPROVED** [review]

File: pkg/utils/perm_windows.go [hunk 2]
```diff
@@ -30,16 +31,38 @@ type FileOwner struct {
 // Note: Very unlikely for this to panic if this function is unable to get the current
 // user. Not being able to get the current user, is a critical problem and nothing
 // can continue so a panic is appropriate.
```
Comment (ycombinator) [inline pkg/utils/perm_windows.go:33]: Looks like we no longer `panic` so this comment should be updated?

**Review by andrzej-stencel — APPROVED** [review]

**Comment by michalpristas** [comment]
@blakerouse can we merge this?

File: pkg/utils/perm_windows.go (hunk 2 above)
Comment (blakerouse) [reply pkg/utils/perm_windows.go]: I have updated this comment.

**Review by ycombinator — APPROVED** [review]

**Comment by ycombinator** [comment]
Will force merge once CI is green to work around SonarQube quality gate failure ([26.4% Coverage on New Code](https://sonar.elastic.dev/component_measures?id=elastic_elastic-agent_AYluowg0xMq8P7b4moiZ&pullRequest=5118&metric=new_coverage&view=list) (required ≥ 40%)).

**Comment by ycombinator** [comment]
The `TestEndpointAgentServiceMonitoring/TestEndpointMetricsAfterRestart` integration test seems to be failing consistently.

**Comment by pierrehilbert** [comment]
`TestEndpointAgentServiceMonitoring/TestEndpointMetricsAfterRestart` is not failing because of this change (we have the same failure on main and on other PRs).
I reached out the Endpoint team to check if something changed on their end and will update here when I have more information.

**Comment by ycombinator** [comment]
> `TestEndpointAgentServiceMonitoring/TestEndpointMetricsAfterRestart` is not failing because of this change (we have the same failure on main and on other PRs).

Thanks for checking, @pierrehilbert.  In that case, I think we can go ahead and merge this PR?

**Comment by pierrehilbert** [comment]
Now the new Endpoint build is available, tests are passing and we should be good to merge here.

**Comment by cmacknz** [comment]
Force merging past the test coverage requirement.

**Pull request merged**